src/embed.js

```diff
@@ -177,7 +175,10 @@ function embed(el, spec, opt, callback) {
           .text('Open in Vega Editor')
           .attr('href', '#')
           .on('click', function() {
-            post(window, embed.config.editor_url, {spec: source, mode: mode});
+            post(window, embed.config.editor_url, {
+              spec: JSON.stringify(spec, null, 2),
```

Comment:
perhaps.  In the last patch, i simply rearranged the existing code to only do stringify when actually needed. Feel free to remove it outright :)